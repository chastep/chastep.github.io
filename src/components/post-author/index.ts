export { PostAuthor } from "./post-author";
