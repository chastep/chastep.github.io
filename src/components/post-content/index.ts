export { PostContent } from "./post-content";
