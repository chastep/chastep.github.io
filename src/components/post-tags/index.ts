export { PostTags } from "./post-tags";