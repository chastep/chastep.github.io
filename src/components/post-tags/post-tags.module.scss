@use "src/assets/styles/mixins";
@use "src/assets/styles/variables";

.post-tags {
  list-style: none;
  @include mixins.margin-bottom(0.5);
  @include mixins.padding-equal(0);

  .item {
    display: inline-block;
    @include mixins.margin(0.25, 0);

    @include mixins.breakpoint-sm {
      &:first-child {
        @include mixins.margin-left(0);
        @include mixins.padding-left(0);
      }
    }
  }
}
