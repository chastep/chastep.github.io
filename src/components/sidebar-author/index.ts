export { SidebarAuthor } from "./sidebar-author";