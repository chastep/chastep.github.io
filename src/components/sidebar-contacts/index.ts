export { SidebarContacts } from "./sidebar-contacts";
