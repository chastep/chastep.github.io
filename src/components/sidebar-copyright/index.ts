export { SidebarCopyright } from "./sidebar-copyright";
