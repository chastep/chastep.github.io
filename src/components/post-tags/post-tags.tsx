import React, { type FC } from "react";

import { Button } from "@/components/button";

import * as styles from "./post-tags.module.scss";

interface PostTagsProps {
  tags: string[];
  tagSlugs: string[];
};

const PostTags: FC<PostTagsProps> = ({ tags, tagSlugs }) => (
  <ul className={styles.postTags}>
    {tagSlugs.map((slug, i) => (
      <li className={styles.item} key={slug}>
        <Button title={tags[i]} key={slug} to={slug} />
      </li>
    ))}
  </ul>
);

export { PostTags };
