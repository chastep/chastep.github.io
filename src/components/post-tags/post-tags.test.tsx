import React from "react";
import { test, describe, expect } from "bun:test";
import { render as reactTestingLibraryRender } from "@testing-library/react";

import * as mocks from "@/mocks";
import { PostTags } from "@/components/post-tags";
import { createSnapshotsRenderer } from "@/utils/render-with-coil-provider";

describe("PostTags", () => {
  test("renders correctly", () => {
    const props = {
      tags: mocks.markdownRemark.frontmatter.tags,
      tagSlugs: mocks.markdownRemark.fields.tagsSlugs,
    };

    const tree = createSnapshotsRenderer(<PostTags {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("buttons is rendered correctly and exists", () => {
    const props = {
      tags: mocks.markdownRemark.frontmatter.tags,
      tagSlugs: mocks.markdownRemark.fields.tagsSlugs,
    };

    const tree = reactTestingLibraryRender(<PostTags {...props} />);

    props.tags.forEach((tag) => {
      expect(tree.getByText(tag)).toBeInTheDocument();
    });
  });
});
