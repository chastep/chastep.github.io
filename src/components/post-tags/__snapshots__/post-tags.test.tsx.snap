// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`PostTags renders correctly 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    {
      "$$typeof": Symbol(react.test.json),
      "children": [
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            "Handwriting",
          ],
          "props": {
            "aria-current": undefined,
            "className": "",
            "href": "/handwriting",
            "onClick": [Function],
            "onMouseEnter": [Function],
          },
          "type": "a",
        },
      ],
      "props": {
        "className": undefined,
      },
      "type": "li",
    },
    {
      "$$typeof": Symbol(react.test.json),
      "children": [
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            "Helvetica",
          ],
          "props": {
            "aria-current": undefined,
            "className": "",
            "href": "/helvetica",
            "onClick": [Function],
            "onMouseEnter": [Function],
          },
          "type": "a",
        },
      ],
      "props": {
        "className": undefined,
      },
      "type": "li",
    },
  ],
  "props": {
    "className": undefined,
  },
  "type": "ul",
}
`;
